### ISSUE TITLE
Feature Request: add option to `no-raw-locators` to allow a locator as an argument

### ISSUE THREAD

**Issue by heygarrett** [playwright-community/eslint-plugin-playwright#373]
The raw locator method accepts either a selector or a locator as an argument, and thanks to #189 there's an option to create an allow list for specific selectors. But it would be great if there was also an option to generally allow a locator as an argument to the raw locator method.

**Comment by MSroczynski3** [comment]
Hi @heygarrett !
Could you please elaborate what do you mean by this statement?
> an option to generally allow a locator as an argument

It would be ideal if you could provide an example or two. I could then give a try coding it 😄 🚀

**Comment by heygarrett** [comment]
@MSroczynski3 Sure! Here's a contrived example: If I have a page element that's reused in a few places I'll create a generic locator for it. And then I'll try to locate specific instances of it by passing that generic locator to the `locator()` method:

```javascript
const button = page.getByRole("button", { name: "common button" });

const firstButton = page.getByRole("region", { name: "first" }).locator(button);
const secondButton = page.getByRole("region", { name: "second" }).locator(button);
```

I'd love an option to allow this in the `no-raw-locator` rule.

**Comment by mskelton** [comment]
@heygarrett This is not valid Playwright code given the types:

https://github.com/microsoft/playwright/blob/main/packages/playwright-core/src/client/page.ts#L672C3-L672C10

This works only by the fact that locators have a `toString` method thus they are coerced properly. I'm going to close this as won't fix given the ESLint plugin should not promote incorrect Playwright code.

**Comment by heygarrett** [comment]
@mskelton That's only for `Page.locator()`; `Locator.locator()` accepts a selector or a locator:  
https://playwright.dev/docs/api/class-locator#locator-locator

**Comment by mskelton** [comment]
@heygarrett Oops, you're right, sorry about that! Merged the PR to fix.
